### PR TITLE
Validate cron query on add (SYN-5364)

### PR DESCRIPTION
**Pull request by mikemoritz**
Also checked that trigger and dmon queries were validated.

### CODE REVIEW

File: synapse/tests/test_lib_agenda.py [hunk 1]
```diff
@@ -407,17 +413,13 @@ async def test_agenda_persistence(self):
         ''' Test we can make/change/delete appointments and they are persisted to storage '''
         with self.getTestDir() as fdir:
 
-            core = mock.Mock()
+            core = mock.AsyncMock()
 
-            def raiseOnBadStorm(q):
+            async def raiseOnBadStorm(q):
                 ''' Just enough storm parsing for this test '''
-                # TODO: Async this and use AsyncMock when Python 3.8+ only
```
Comment (vEpiphyte) [inline synapse/tests/test_lib_agenda.py:414]: this is still a valid TODO and something we can act on now

File: synapse/tests/test_cortex.py [hunk 2]
```diff
@@ -5968,6 +5968,9 @@ async def test_cortex_storm_lib_dmon_cmds(self):
             msgs = await core.stormlist('dmon.list')
             self.stormIsInPrint('(wootdmon            ): error', msgs)
 
+            # invalid storm query
+            await self.asyncraises(s_exc.BadSyntax, core.nodes('$lib.dmon.add(" | | | ")'))
```
Comment (vEpiphyte) [inline synapse/tests/test_cortex.py:5972]: if we use self.raises() over self.asyncraises the context manager is also inspectable

File: synapse/tests/test_lib_agenda.py (hunk 1 above)
Comment (mikemoritz) [reply synapse/tests/test_lib_agenda.py:414]: The function was async'd and the mock changed to AsyncMock as part of this PR.

File: synapse/tests/test_cortex.py (hunk 2 above)
Comment (mikemoritz) [reply synapse/tests/test_cortex.py:5972]: I didn't see a need to inspect the context manager for this part of the test

**Review by vEpiphyte — APPROVED** [review]

**Pull request merged**